deprecates resetAllAttrs and resetLocalAttrs in favor of the new API

xeno-by opened the pull request:
We now have c.untypecheck, which is supposed to be a counterpart of c.typecheck
in the sense that it goes back from typed trees to untyped ones:
http://stackoverflow.com/questions/20936509/scala-macros-what-is-the-difference-between-typed-aka-typechecked-an-untyped.

Let’s hope that c.untypecheck will soon be able to solve our problems
with partially/incorrectly attributed trees emitted by macros:
https://groups.google.com/forum/#!topic/scala-internals/TtCTPlj_qcQ.

xeno-by committed:
deprecates resetAllAttrs and resetLocalAttrs in favor of the new API

We now have c.untypecheck, which is supposed to be a counterpart of c.typecheck
in the sense that it goes back from typed trees to untyped ones:
http://stackoverflow.com/questions/20936509/scala-macros-what-is-the-difference-between-typed-aka-typechecked-an-untyped.

Let’s hope that c.untypecheck will soon be able to solve our problems
with partially/incorrectly attributed trees emitted by macros:
https://groups.google.com/forum/#!topic/scala-internals/TtCTPlj_qcQ.

xeno-by commented:
review @retronym

xeno-by commented:
Please feel free to decide whether this should go into M8 or RC1.

densh commented on src/reflect/scala/reflect/macros/Typers.scala:
```diff
@@ -69,18 +69,33 @@ trait Typers {
+   *
+   *  Read http://stackoverflow.com/questions/20936509/scala-macros-what-is-the-difference-between-typed-aka-typechecked-an-untyped
+   *  for more information.
+   */
```
Is there any way to avoid copy-paste here? If not, a side-comment that somehow textually links the two would be helpful.

xeno-by replied:
Not sure whether there are scaladoc macros that can span multiple modules. @heathermiller @VladUreche could you help here?

VladUreche replied:
Nope, they can't :-1: Sorry :(

retronym replied:
How about a link ("more details in ...") from the documentation one method to the docs of the other?

xeno-by replied:
Okay, we could link the toolbox doc (arguably, much less important) to the context doc. Will do now.

VladUreche replied:
That would work, thanks Jason! Btw, the last line:

```
   *  Read http://stackoverflow.com/questions/20936509/scala-macros-what-is-the-difference-between-typed-aka-typechecked-an-untyped
   *  for more information.
```

could also be written as:

```
   *  @see [[http://stackoverflow.com/questions/20936509/scala-macros-what-is-the-difference-between-typed-aka-typechecked-an-untyped]]
```

xeno-by commented:
@retronym @densh Addresses the feedback

xeno-by committed:
moves analyzer.ImportType into scala.reflect.internal

This cute little type is necessary for importers to work correctly.
I wonder how we could overlook its existence for almost 2 years.

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
(kitty-note-to-self: ignore 32096705)
:cat: Roger! Rebuilding pr-scala for 917de78d, db623cf0. :rotating_light:

retronym commented:
PLS REBUILD ALL

scala-jenkins commented:
(kitty-note-to-self: ignore 32246131)
:cat: Roger! Rebuilding pr-scala for 917de78d, db623cf0. :rotating_light:

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
(kitty-note-to-self: ignore 32529856)
:cat: Roger! Rebuilding pr-scala for 917de78d, db623cf0. :rotating_light:

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
(kitty-note-to-self: ignore 32759335)
:cat: Roger! Rebuilding pr-scala for 917de78d, db623cf0. :rotating_light:

xeno-by commented:
This PR seems to be particularly unlucky with the kitten. I'll close it and reopen it again.

xeno-by commented:
https://github.com/scala/scala/pull/3392
